src/timer/mod.rs: method not found in `&mut u16 in pcs register.

I am trying to see if I can fix #2672 compile issue on the `STM32L4`.
So I modified the [stm32-data](https://github.com/vDorst/stm32-data/tree/octospi), generated the data.
Now I compiled `embassy-stm32`, which usage 963fda240484e0105408185f955703e2c02f9187 as base also has pull #2672 in it, with my own generated data. build cmd: `cargo build -F stm32l4s5zi,memory-x --target thumbv7em-none-eabihf`.

Octospi error's are not showing up any more now I am getting errors about a timer.
It seems related to latest commit https://github.com/embassy-rs/stm32-data/commit/f84633553331c2d154ee72de779a40cbb10fd1bd.

```
# cargo build -F stm32l4s5zi,memory-x --target thumbv7em-none-eabihf
   Compiling embassy-stm32 v0.1.0 (/home/r/dev/rust/embassy/embassy-stm32)

error[E0599]: no method named `set_psc` found for mutable reference `&mut u16` in the current scope
   --> src/timer/mod.rs:100:36
    |
100 |             regs.psc().write(|r| r.set_psc(psc));
    |                                    ^^^^^^^ method not found in `&mut u16`

error[E0599]: no method named `psc` found for type `u16` in the current scope
   --> src/timer/mod.rs:140:41
    |
140 |             let psc = regs.psc().read().psc();
    |                       ----              ^^^ method not found in `u16`
    |                       |
    |                       method `psc` is available on `TimCore`

error[E0599]: no method named `set_psc` found for mutable reference `&mut u16` in the current scope
   --> src/timer/mod.rs:381:36
    |
381 |             regs.psc().write(|r| r.set_psc(psc));
    |                                    ^^^^^^^ method not found in `&mut u16`

error[E0599]: no method named `psc` found for type `u16` in the current scope
   --> src/timer/mod.rs:395:41
    |
395 |             let psc = regs.psc().read().psc();
    |                       ----              ^^^ method not found in `u16`
    |                       |
    |                       method `psc` is available on `TimGp32`

For more information about this error, try `rustc --explain E0599`.

```

rust 1.75

--- a/embassy-stm32/Cargo.toml
+++ b/embassy-stm32/Cargo.toml
@@ -70,7 +70,7 @@ rand_core = "0.6.3"
 sdio-host = "0.5.0"
 critical-section = "1.1"
 #stm32-metapac = { version = "15" }
-stm32-metapac = { git = "https://github.com/embassy-rs/stm32-data-generated", tag = "stm32-data-c8b32ecae7d70cea2705095c4fc6bd5f59d238d5" }
+stm32-metapac = { git = "https://github.com/embassy-rs/stm32-data-generated", tag = "stm32-data-f84633553331c2d154ee72de779a40cbb10fd1bd" }
 vcell = "0.1.3"
 nb = "1.0.0"
 stm32-fmc = "0.3.0"
@@ -94,7 +94,7 @@ critical-section = { version = "1.1", features = ["std"] }
 proc-macro2 = "1.0.36"
 quote = "1.0.15"
 #stm32-metapac = { version = "15", default-features = false, features = ["metadata"]}
-stm32-metapac = { git = "https://github.com/embassy-rs/stm32-data-generated", tag = "stm32-data-c8b32ecae7d70cea2705095c4fc6bd5f59d238d5", default-features = false, features = ["metadata"]}
+stm32-metapac = { git = "https://github.com/embassy-rs/stm32-data-generated", tag = "stm32-data-f84633553331c2d154ee72de779a40cbb10fd1bd", default-features = false, features = ["metadata"]}
 
 
 [features]

--- a/embassy-stm32/src/time_driver.rs
+++ b/embassy-stm32/src/time_driver.rs
@@ -286,7 +286,7 @@ impl RtcDriver {
             Ok(n) => n,
         };
 
-        r.psc().write(|w| w.set_psc(psc));
+        r.psc().write_value(psc);
         r.arr().write(|w| w.set_arr(u16::MAX));
 
         // Set URS, generate update and clear URS

--- a/embassy-stm32/src/timer/mod.rs
+++ b/embassy-stm32/src/timer/mod.rs
@@ -97,7 +97,7 @@ pub(crate) mod sealed {
             let arr = unwrap!(u16::try_from(divide_by - 1));
 
             let regs = Self::regs_core();
-            regs.psc().write(|r| r.set_psc(psc));
+            regs.psc().write_value(psc);
             regs.arr().write(|r| r.set_arr(arr));
 
             regs.cr1().modify(|r| r.set_urs(vals::Urs::COUNTERONLY));
@@ -137,7 +137,7 @@ pub(crate) mod sealed {
 
             let regs = Self::regs_core();
             let arr = regs.arr().read().arr();
-            let psc = regs.psc().read().psc();
+            let psc = regs.psc().read();
 
             timer_f / arr / (psc + 1)
         }
@@ -378,7 +378,7 @@ pub(crate) mod sealed {
             let arr: u32 = unwrap!((pclk_ticks_per_timer_period / (psc as u64 + 1)).try_into());
 
             let regs = Self::regs_gp32();
-            regs.psc().write(|r| r.set_psc(psc));
+            regs.psc().write_value(psc);
             regs.arr().write_value(arr);
 
             regs.cr1().modify(|r| r.set_urs(vals::Urs::COUNTERONLY));
@@ -392,7 +392,7 @@ pub(crate) mod sealed {
 
             let regs = Self::regs_gp32();
             let arr = regs.arr().read();
-            let psc = regs.psc().read().psc();
+            let psc = regs.psc().read();
 
             timer_f / arr / (psc + 1)
         }

